src/Microsoft.Restier.Security/ApiConfigurationExtensions.cs

```diff
@@ -32,8 +32,8 @@ public static class ApiConfigurationExtensions
             this ApiBuilder builder)
         {
             Ensure.NotNull(builder, "builder");
-            builder.AddHookHandler<IQueryExpressionInspector>(RoleBasedAuthorization.Default);
-            builder.AddHookHandler<IQueryExpressionExpander>(RoleBasedAuthorization.Default);
+            builder.CutoffPrevious<IQueryExpressionInspector, RoleBasedAuthorization>();
+            builder.ChainPrevious<IQueryExpressionExpander, RoleBasedAuthorization>();
```

Comment:
I agree. There were so many classes in RESTier that took multiple duties (hook handlers). That makes the logic unclear. IMHO even if we have something to share among the "logically" two hook handlers, we shall seek another way to achieve that (such as putting shared state into DI). Does it make sense?